tools/link.py

```diff
@@ -360,6 +360,9 @@ def get_binaryen_passes(memfile):
   # hardcoded value in the binaryen pass)
   if optimizing and settings.GLOBAL_BASE >= 1024:
```

Comment:
Perhaps instead you want to add `and not  settings.STACK_FIRST` to this condition?

Reply:
Thanks, good idea, I see now that indeed we already parse this, so makes sense to read it from there. Done.